modules/react-native-status/desktop/rctstatus.cpp

```diff
@@ -29,10 +29,18 @@ struct RegisterQMLMetaType {
 
 class RCTStatusPrivate {
 public:
-    Bridge* bridge = nullptr;
+    static Bridge* bridge;
+    static RCTStatus* rctStatus;
 };
 
-RCTStatus::RCTStatus(QObject* parent) : QObject(parent), d_ptr(new RCTStatusPrivate) {}
+Bridge* RCTStatusPrivate::bridge = nullptr;
+RCTStatus* RCTStatusPrivate::rctStatus = nullptr;
+
+RCTStatus::RCTStatus(QObject* parent) : QObject(parent), d_ptr(new RCTStatusPrivate) {
+    RCTStatusPrivate::rctStatus = this;
+    SetJailSignalCallback((void*)&RCTStatus::jailSignalEventCallback);
```

Comment:
@MaxRis, maybe we can use `SetJailSignalCallback((void*)&RCTStatus::jailSignalEvent);` ?
If that will work, we can get rid of two functions: `jailSignalEventCallback` and `emitSignalEvent`

Reply:
Probably, it's possible to optimize current solution by passing non-static function pointer instead of static.

> maybe we can use SetJailSignalCallback((void*)&RCTStatus::jailSignalEvent); ?
I afraid that this will break current implementation details:
- bridge->eventDispatcher()->sendDeviceEvent("gethEvent", QVariantMap{{"jsonEvent", signal}}); ends with writing to QTcpSocket(if I'm right here) and it seats at main thread. Access of QTcpSocket from non main, status-go thread breaks some internal flow of QTcpSocket. So, I've used signal/slot connection to transmit jail event to main thread.

Reply:
Ah, missed that `jailSignalEventCallback` is static, thanks for explanation!